distributed/scheduler.py

```diff
@@ -3425,6 +3429,39 @@ def worker_objective(self, ts: TaskState, ws: WorkerState) -> tuple:
         else:
             return (start_time, ws._nbytes)
 
+    @ccall
+    def add_replica(self, ts: TaskState, ws: WorkerState):
+        """Note that a worker holds a replica of a task with state='memory'"""
+        if self._validate:
+            assert ws not in ts._who_has
+            assert ts not in ws._has_what
+
+        ws._nbytes += ts.get_nbytes()
```

Comment:
Perhaps some mysterious magic type annotations are necessary to get this function to Cythonize well? I have no idea.

Reply:
Could you elaborate on why you think it's not cythonizing well? I'm trying to build the HTML report but failing:

```

$ python -m pip install -vv --no-deps --install-option="--with-cython" -e .
(works, but doesn't generate the HTML)
$ cythonize -a distributed/scheduler.py
    @ccall
    @inline
    @nogil
    def get_nbytes(self) -> Py_ssize_t:
                  ^
------------------------------------------------------------
distributed/scheduler.py:1638:19: Operation not allowed without gil
```

Reply:
Added explicit declaration of nbytes.
I also noticed that replacing 
```python
        ws._nbytes -= ts.get_nbytes()
        del ws._has_what[ts]
        ts._who_has.remove(ws)
        if len(ts._who_has) == 1:
            self._replicated_tasks.remove(ts)
```
with
```python
        wh: set = ts._who_has
        hw: dict = ws._has_what
        ws._nbytes -= ts.get_nbytes()
        del hw[ts]
        wh.remove(ws)
        if len(wh) == 1:
            rt: set = self._replicated_tasks
            rt.remove(ts)
```
marginally improves the Cython code, at the cost of greatly reduced readability, so I'd rather not do it.